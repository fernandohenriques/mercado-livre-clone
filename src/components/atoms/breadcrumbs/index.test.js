import React from 'react';
import { shallow } from 'enzyme';
import { render } from '@testing-library/react';

import Breadcrumbs from './index';

describe('<Breadcrumbs />', () => {
  it('Snapshot testing', () => {
    const { asFragment } = render(
      <Breadcrumbs categories={['Eletronica y Audio', 'iPod', 'reproductores', '32 GB']} />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('Should render the last item with strong tag and others with span tag', () => {
    const lastItem = '32 GB';
    const wrapper = shallow(
      <Breadcrumbs categories={['Eletronica y Audio', 'iPod', 'reproductores', 'iPod touch', lastItem]} />
    );

    expect(wrapper.exists()).toBeTruthy();
    expect(wrapper.find('span')).toHaveLength(4);
    expect(wrapper.find('strong')).toHaveLength(1);
    expect(wrapper.find('strong').text()).toEqual(lastItem);
  });
});
