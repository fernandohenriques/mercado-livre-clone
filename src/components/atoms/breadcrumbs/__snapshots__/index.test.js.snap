// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Breadcrumbs /> Snapshot testing 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <span>
      Eletronica y Audio 
      <div />
       
    </span>
    <span>
      iPod 
      <div />
       
    </span>
    <span>
      reproductores 
      <div />
       
    </span>
    <strong>
      32 GB
    </strong>
  </div>
</DocumentFragment>
`;
